src/components/LabelPicker/__tests__/LabelPicker.spec.tsx

```diff
@@ -0,0 +1,340 @@
+import * as React from 'react';
+import { act } from 'react-dom/test-utils';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import userEvent from '@testing-library/user-event';
+import { LabelPicker } from '../LabelPicker';
+import { Button } from '../../Buttons';
+import { useDocumentEvent } from '../../../utils/testUtils';
+
+const labelsMock = [
+    {
+        name: 'First label',
+        isSelected: false,
+    },
+    {
+        name: 'Second label',
+        isSelected: true,
+        count: 12,
+    },
+    {
+        name: 'Third label',
+        isSelected: false,
+        count: 0,
+    },
+    {
+        name: 'Fourth label',
+        isSelected: false,
+        count: 0,
+        id: 'random',
+    },
+];
+
+const defaultProps = {
+    labels: [],
+    doneLabel: '',
+    inputPlaceholder: '',
+    onAdd: () => {},
+    onChange: () => {},
+};
+
+let view;
+
+const renderLabelPicker = () => {
+    view = render(
+        <LabelPicker {...defaultProps}>
+            <Button>Click me</Button>
+        </LabelPicker>
+    );
+};
+
+const rerenderLabelPicker = (props) => {
+    view = render(
+        <LabelPicker {...defaultProps} {...props}>
+            <Button>Click me</Button>
+        </LabelPicker>
+    );
+};
```

Comment:
beforeEach can handle the first render which renders the default component and using that we don't need to call in every test renderLabelPicker and looks cleaner IMO.

calling rerenderLabelPicker when it doesn't rerender can be misleading, because here it is a normal render. If it is a rerender we should use `rerender` instead of `render`. 

In the tests where we need to pass different props we can just rerender the component with the desired props using `rerender`.

Reply:
:+1: